### PR TITLE
EMSUSD-255 Fixes an issue in Layer Editor where Save All icon was not functioning properly.

### PR DESCRIPTION
**Pull request by AramAzhari-adsk**
EMSUSD-255 Fixes an issue in Layer Editor where Save All icon was not functioning properly.

The issue occurred due to Qt UI layout changes. The solution is to keep track of the SaveLayerPathRow data separately so that it can be iterated over when saving all layers.

### CODE REVIEW

File: lib/usd/ui/layerEditor/saveLayersDialog.cpp [hunk 1]
```diff
@@ -632,13 +635,8 @@ QString SaveLayersDialog::buildTooltipForLayer(SdfLayerRefPtr layer)
...
 QWidget* SaveLayersDialog::findEntry(SdfLayerRefPtr key)
 {
-    if (!_anonLayersWidget || !_anonLayersWidget->layout()) {
-        return nullptr;
-    }
-
-    QLayout* anonLayout = _anonLayersWidget->layout();
-    for (int i = 0, count = anonLayout->count(); i < count; ++i) {
-        auto row = dynamic_cast<SaveLayerPathRow*>(anonLayout->itemAt(i)->widget());
+    for (int i = 0, count = _saveLayerPathRows.size(); i < count; ++i) {
```
Comment (AramAzhari-adsk) [inline lib/usd/ui/layerEditor/saveLayersDialog.cpp:638]: I kept the order of iteration the same as before, Although in this case it shouldn't matter as it is only picking one item.

File: lib/usd/ui/layerEditor/saveLayersDialog.cpp [hunk 2]
```diff
@@ -649,13 +647,8 @@ QWidget* SaveLayersDialog::findEntry(SdfLayerRefPtr key)
...
 void SaveLayersDialog::forEachEntry(const std::function<void(QWidget*)>& func)
 {
-    if (!_anonLayersWidget || !_anonLayersWidget->layout()) {
-        return;
-    }
-
-    QLayout* anonLayout = _anonLayersWidget->layout();
-    for (int i = 0, count = anonLayout->count(); i < count; ++i) {
-        func(anonLayout->itemAt(i)->widget());
+    for (int i = 0, count = _saveLayerPathRows.size(); i < count; ++i) {
```
Comment (AramAzhari-adsk) [inline lib/usd/ui/layerEditor/saveLayersDialog.cpp:650]: Same here, the order of iteration is from top layer to bottom

**Comment by AramAzhari-adsk** [comment]
The previous preflight builds fully passed, I only added a comment line in the last commit.
Please refer to Build# 1197 for verification.

**Review by pierrebai-adsk — APPROVED** [review]
LGTM

**Pull request merged**